File: 02_Object_Oriented_Programming/02_Class_Relationships/01_Inheritance/python/main.py
"""
Driver — demonstrates Inheritance (IS-A relationship).
Run: python main.py
"""

from dog import Dog


def main() -> None:
    dog = Dog("Buddy", "Golden Retriever")

    dog.eat()    # inherited from Animal
    dog.sleep()  # inherited from Animal
    dog.bark()   # Dog-specific

    print(dog)   # __str__


if __name__ == "__main__":
    main()

"""
Expected Output:
Buddy is eating.
Buddy is sleeping.
Buddy barks: Woof Woof!
Dog(Buddy, breed=Golden Retriever)
"""

File: 02_Object_Oriented_Programming/02_Class_Relationships/01_Inheritance/javascript/main.js
/**
 * Driver — demonstrates Inheritance (IS-A relationship).
 * Run: node main.js
 */

const Dog = require("./dog");

const dog = new Dog("Buddy", "Golden Retriever");

dog.eat();   // inherited from Animal
dog.sleep(); // inherited from Animal
dog.bark();  // Dog-specific

console.log(`${dog}`); // toString

/*
Expected Output:
Buddy is eating.
Buddy is sleeping.
Buddy barks: Woof Woof!
Dog(Buddy, breed=Golden Retriever)
*/
